### ISSUE TITLE
Incorrect table parsing

### ISSUE THREAD

**Issue by knkski** [html5lib/html5lib-python#392]
I've got an HTML document that contains some text inside of a table row:

```html
<!DOCTYPE html>
<html>
  <body>
    <table>
      <tr>foo</tr>
    </table>
  </body>
</html>
```

It looks like the text gets pulled outside of the table element for some reason (not sure how to print it with normal HTML-style syntax, but you can still see that `foo` has been moved):

```python
>>> import html5lib
>>> import xml.etree.ElementTree as ET
>>> ET.tostring(html5lib.parse('<!DOCTYPE html><html><body><table><tr>foo</tr></table></body></html>'))
b'<html:html xmlns:html="http://www.w3.org/1999/xhtml"><html:head /><html:body>foo<html:table><html:tbody><html:tr /></html:tbody></html:table></html:body></html:html>'
```

This is despite the example parsing correctly when using just `xml.etree`:

```python
>>> import xml.etree.ElementTree as ET
>>> ET.tostring(ET.ElementTree(ET.fromstring('<!DOCTYPE html><html><body><table><tr>foo</tr></table></body></html>')).getroot())
b'<html><body><table><tr>foo</tr></table></body></html>'
```

This doesn't happen with other elements such as `<div>`. This was using Python 3.6 and html5lib 1.0.1.

**Comment by gsnedders** [comment]
This is what the HTML spec says, and this is what browsers do. (This is called "foster parenting" in the spec, if you're curious.)

**Comment by gsnedders** [comment]
Ah, right, that's us still not having implemented `template` yet, which is part of #56.
